Format with clang-format in future

markuspg opened the pull request:
This pull request does not introduce any functional changes to the code. I just wanted to propose to format the code automatically with `clang-format` in future. For this reason I added a **default** `.clang-format` file and formatted the code accordingly.

I tried to create a clang-format configuration which was better suited to the currently existing codebase than the default one. But the handling of tabs proved to be very difficult, especially for continued function parameter lists.

I guess this pull request is not for merging directly, but rather for discussion.

If there is an automatic formatting in place already and I just overlooked it, this pull request can be rejected gladly.

PS: This is not about cheaply making me the "king of `git blame`". Any port to a newer GTK version or another toolkit should touch most lines anyhow and flush these changes down the Git history.

TobiX commented:
Good idea - I think it would be fine to migrate to some "standard" style, even if that makes the diff a lot larger (would it?). You can always add a `.git-blame-ignore-revs` file later (see https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view)

markuspg committed:
Add clang-format v7 default configuration

markuspg committed:
Format codebase according to .clang-format def

markuspg committed:
Add .git-blame-ignore-revs file

markuspg commented:
The `.clang-format` I added is the default when calling `clang-format --dump-config` (i.e. the _LLVM_ style). My manually adjusted one was much smaller. But the result was a bad compromise in any case, so I prefer going with the default of `clang-format`.

One change I conducted is creating the `.clang-format` file from `clang-format 7` instead of `clang-format 17`. This maintains backwards compatibility down to _Debian 10_. Despite quite a few options not existing in the earlier version, the resulting formatting is identical no matter if `clang-format 7` or `clang-format 17` is being used with the `.clang-format` file.

Thank you for the hint for the `.git-blame-ignore-revs` file, that's a really neat feature. I added the file. Unfortunately I didn't find a way to have `git blame` pick it up automatically on any checkout. But at least the file exists as a hint at this feature.

TobiX commented:
> Unfortunately I didn't find a way to have `git blame` pick it up automatically on any checkout. But at least the file exists as a hint at this feature.

Yeah, it's unfortunate one can't just `git config --global blame.ignoreRevsFile .git-blame-ignore-revs`, because then `git blame` complains in repositories not containing this file. I built some automation in my own shell config, if yo need some inspiration: https://github.com/TobiX/dotfiles/commit/9d1674c4f49cac050ad4000b25f2dabd131cdb67
